### PR TITLE
Feat worker stop

### PR DESCRIPTION
**Pull request by abnegate**
Allow hooking worker stop events to reclaim worker-level resources

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Build image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: php-${{ matrix.php-versions }}.Dockerfile
+          push: false
+          tags: websocket-dev
+          load: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```
Comment (lohanidamodar) [inline .github/workflows/tests.yml:39]: I don't think this is required
I use this in all repositories?
https://github.com/utopia-php/docs/blob/docs-testing/docs/testing.md#using-docker

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Build & Start Containers
+        run:  |
+          export PHP_VERSION=${{ matrix.php-versions }} 
+          docker compose build
```
Comment (lohanidamodar) [inline .github/workflows/tests.yml:44]: if you do use the buildx above, I don't think we need to build again

File: composer.json [hunk 3]
```diff
@@ -24,7 +26,7 @@
         "phpunit/phpunit": "^9.5.5",
         "vimeo/psalm": "^4.8.1",
```
Comment (lohanidamodar) [inline composer.json:27]: I think we can remove psalm

File: .github/workflows/tests.yml (hunk 1 above)
Comment (abnegate) [reply .github/workflows/tests.yml:39]: Not required but this caches the image so it isn't rebuilt for each run

File: .github/workflows/tests.yml (hunk 2 above)
Comment (abnegate) [reply .github/workflows/tests.yml]: Removed here

File: composer.json (hunk 3 above)
Comment (abnegate) [reply composer.json]: Removed

File: .github/workflows/tests.yml (hunk 1 above)
Comment (abnegate) [reply .github/workflows/tests.yml:39]: Looks like this doesn't work with multi-version tests though, will remove

**Pull request merged**